java/app/src/main/java/com/fullstorydev/shoppedemo/ui/checkout/CheckoutFragment.java

```diff
@@ -60,8 +62,25 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
             if(!isLoading){
                 checkoutViewModel.fetchCustomerInfo();
                 binding.setViewmodel(checkoutViewModel);
-                binding.executePendingBindings();
+                binding.setHandlers(this);
             }
         });
     }
+
+    @Override
+    public void onClickPurchase(CustomerInfo customerInfo, Double subtotal) {
+        try{
+            boolean valid = customerInfo.validateOrder();
+            if(valid && subtotal!= null && subtotal>0) {
```

Comment:
Nit: whitespace around the operator

Reply:
🙏 gonna need to add lint for this project at some point..